Move all remaining frameworks to SwiftPM modules

mokagio opened the pull request:
## Description

- Closes https://linear.app/a8c/issue/AINFRA-511/move-yosemite-tests-to-modules
- Closes https://linear.app/a8c/issue/AINFRA-510/move-hardware-to-modules
- Closes https://linear.app/a8c/issue/AINFRA-473/move-fakes-to-modules
- Closes https://linear.app/a8c/issue/AINFRA-470/move-networking-to-modules
- Closes https://linear.app/a8c/issue/AINFRA-469/move-storage-to-modules
- Closes https://linear.app/a8c/issue/AINFRA-468/move-yosemite-to-modules
- Closes https://linear.app/a8c/issue/AINFRA-672/move-uitestsfoundation-to-modules

Moves the following frameworks and their tests to Swift packages in `Modules/`:

- Networking
- Storage
- Hardware
- Fakes
- UITestsFoundation

The only framework that hasn't been moved is WordPressAuthenticator because of circular Obective-C-Swift dependencies. Addressing that is out of scope.

<!-- Id number of the Linear issue this PR addresses. -->
<!-- Part of: # -> use this if your PR is one of many related to one issue -->

<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

## Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->

## Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

dangermattic commented:
<!--
  0 Errors
  2 Warnings: View files have been modified,...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Simplify Networking into a Swift package in Modules - Tests fail

Tests fail because of Fakes, which hasn't been migrated yet.

mokagio committed:
Point Aztec to version that export UIKit-less HTMLParser

This is required to make the watchOS app compile with the new setup.
Before, the code relied on `canImport(Aztec)`, but that check always
passes in SwiftPM.

mokagio committed:
AINFRA-469 - Simplify Storage into a Swift package

mokagio committed:
AINFRA-510 - Simplify Hardware into a Swift module

mokagio committed:
AINFRA-468 - Simplify Yosemite into a Swift package - Tests fail

Tests fail because of the dependency on Fakes

mokagio committed:
AINFRA-473 - Simply Fakes into a Swift package

mokagio committed:
Brute force `@testable import NetworkingCore` in all Networking tests

It was simpler than iterating build after build to identify the ones
that needed it.

mokagio committed:
Track `Package.resolved` change—however unsure of source

mokagio committed:
AINFRA-672 - Simplify UITestsFoundation into a Swift package

mokagio committed:
Make missing NetworkingCore elements fakeable

mokagio commented on Modules/Sources/Codegen/Sourcery/Copiable/Hardware-Copiable.sourcery.yaml:
```diff
@@ -1,7 +1,8 @@
-    file: ../../../../../Hardware/Hardware.xcodeproj
-    target:
-        name: Hardware
+sources:
+    include:
+        - ../../../../../Modules/Sources/Hardware/
```
See https://github.com/woocommerce/woocommerce-ios/pull/15707 for the motivation of these and all the other changes to the Sourcery configuration files.

mokagio commented on Modules/Sources/Fakes/Networking.generated.swift:
```diff
@@ -47,19 +47,6 @@ extension Networking.APNSDevice {
-            gravatarUrl: .fake()
-        )
-    }
-}
```
All the types removed from here went into NetworkingCore.

mokagio commented on Modules/.swiftpm/xcode/xcshareddata/xcschemes/Hardware.xcscheme:
```diff
@@ -14,10 +15,10 @@
+               BlueprintIdentifier = "Hardware"
+               BuildableName = "Hardware"
+               BlueprintName = "Hardware"
+               ReferencedContainer = "container:">
```
A bit frustrating the diff considers this the Storage scheme being moved and update to Hardware, rather than it being the Hardware scheme being moved.

Or maybe something got lost in creating the schemes for the modules and that's why the diff can't pair them 🤔

mokagio commented on Modules/Sources/NetworkingCore/Validators/PlaceholderDataValidator.swift:
```diff
@@ -0,0 +1,11 @@
+    public func validate(data: Data) throws {
+        // no-op
+    }
+}
```
This looks like a new file, but existed already. I guess the new `init` and the various `public` are too much for the diff engine.

https://github.com/woocommerce/woocommerce-ios/blob/ae05dab3f00e2be53a264c2cbbd641a0a5630f5e/Networking/Sources/Core/Validators/PlaceholderDataValidator.swift#L1-L9

mokagio commented on Modules/Tests/StorageTests/Tools/FileStorageTests.swift:
```diff
@@ -10,7 +10,7 @@ final class FileStorageTests: XCTestCase {
         super.setUp()
         subject = PListFileStorage()
-        fileURL = Bundle(for: FileStorageTests.self)
+        fileURL = Bundle.module
```
Uses `.module` instead of `.storage` because in this case we want to load from the test module, not the Storage module under test.

mokagio commented on WooCommerce/WooCommerceTests/Constants.swift:
```diff
@@ -0,0 +1,11 @@
+    /// Default Expectation Timeout
+    ///
+    static let expectationTimeout = TimeInterval(10)
+}
```
Was referenced from Yosemite. Duplicated it for clarity.

mokagio committed:
Remove an unnecessary import

mokagio commented on Modules/Sources/Networking/Extensions/URLRequest+Request.swift:
```diff
@@ -0,0 +1,9 @@
+    public func responseDataValidator() -> ResponseDataValidator {
+        PlaceholderDataValidator()
+    }
+}
```
Looks like a new file but is a modification of an existing one:

https://github.com/woocommerce/woocommerce-ios/blob/ae05dab3f00e2be53a264c2cbbd641a0a5630f5e/Networking/Sources/Extended/Extensions/URLRequest%2BRequest.swift#L1-L8

jaclync committed:
Remove unused `POSEligibilityCheckerProtocol` use case in `BetaFeaturesConfigurationViewModel`.

mokagio commented:
All the frameworks and their tests have been moved to `Modules/`. Closing.
